app/models/spree/adjustment.rb

```diff
@@ -44,7 +44,7 @@ class Adjustment < ApplicationRecord
 
     belongs_to :tax_rate, -> { where spree_adjustments: { originator_type: 'Spree::TaxRate' } },
                foreign_key: 'originator_id'
-    belongs_to :voucher, -> { where spree_adjustments: { originator_type: 'Spree::Voucher' } },
+    belongs_to :voucher, -> { where spree_adjustments: { originator_type: 'Voucher' } },
```

Comment:
If this wasn't caught before then we are missing a spec. Seems a very important part to have a spec for.

Reply:
There is a spec for it : https://github.com/rioug/openfoodnetwork/blob/c4dde06df8d65848f5d112e4f1355df803f90021/spec/models/spree/adjustment_spec.rb#L21 but I am not sure why it's wasn't failing.

Reply:
I'm not sure what exactly the Shoulda matchers test but it seems to fail to verify the `where` clause. It probably just checks that there is an association.

Hm, some testing suggests that it doesn't save any records to the database. It tests only in memory.

I just tried to test in the console and I got a weird error:
```
v = create(:voucher, code: "test")
a = v.create_adjustment("label", create(:order))
a.voucher
ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "spree_adjustments"
LINE 1: ..."deleted_at" IS NULL AND "vouchers"."id" = $1 AND "spree_adj...
                                                             ^

from /home/maikel/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/postgresql_adapter.rb:837:in `prepare'
Caused by PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "spree_adjustments"
LINE 1: ..."deleted_at" IS NULL AND "vouchers"."id" = $1 AND "spree_adj...
                                                             ^

from /home/maikel/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/activerecord-7.0.4.3/lib/active_record/connection_adapters/postgresql_adapter.rb:837:in `prepare'
```

Reply:
This `belongs_to` isn't actually needed, I added it to be consistent with the `tax_rate` adjustment. It turns out the `tax_rate` `belongs_to` is also broken and isn't needed. I dug a little bit into why it wasn't caught by testing, and came across this: https://github.com/thoughtbot/shoulda-matchers/issues/981. `belongs_to` association with custom relation can't be tested with `shoulda-matcher`, that explains the false positive.